filament/src/VertexBuffer.cpp

```diff
@@ -193,4 +200,52 @@ void VertexBuffer::setBufferAt(Engine& engine, uint8_t bufferIndex,
             std::move(buffer), byteOffset, byteSize);
 }
 
+void VertexBuffer::populateTangentQuaternions(const QuatTangentContext& ctx) {
+    if (!ASSERT_PRECONDITION_NON_FATAL(ctx.normals, "Normals must be provided")) {
+        return;
+    }
+
+    // Define a small lambda that converts fp32 into the desired output format.
+    const QuatType quatType = ctx.quatType;
+    const auto writeQuat = [quatType] (quatf inquat, uint8_t* outquat) {
```

Comment:
You should do it the other way around: have each branch of the switch return the proper lambda. This will avoid running a switch on every vertex.

Reply:
Nice!  Done.